lib/manageiq/appliance_console/message_server_configuration.rb

```diff
@@ -0,0 +1,247 @@
+require "manageiq/appliance_console/message_server_configuration/message_server_configuration"
+require "awesome_spawn"
+require "linux_admin"
+require "fileutils"
+
+module ManageIQ
+module ApplianceConsole
+  class MessageServerConfiguration
+    include MessageServerConfiguration
+
+    attr_reader :hostname, :username, :password, :config_dir, :sample_config_dir,
+      :jaas_config_path, :openssl_config_path, 
+      :log4_properties_path, :log4_properties_sample_path, :client_properties_path,
+      :server_properties_path, :server_properties_orig_path,
+      :keystore_dir, :truststore_path, :keystore_path,
+      :messaging_yaml_sample_path, :messaging_yaml_path
+
+    def initialize(options = {})
+      puts("JJV 0.0 #{File.basename(__FILE__)} / #{__method__}")
+      @hostname, @username, @password = nil
+
+      @hostname = get_hostname
+      @username = options[:username] || "admin"
+      @password = options[:password]
+
+      @miq_config_dir              = Pathname.new(MIQ_CONFIG_DIR)
+      @config_dir                  = Pathname.new(CONFIG_DIR)
+      @sample_config_dir           = Pathname.new(SAMPLE_CONFIG_DIR)
+
+      @jaas_config_path            = @config_dir.join("", "kafka_server_jaas.conf")
+      @openssl_config_path         = @config_dir.join("", "openssl_cnf")
+      @log4_properties_path        = @config_dir.join("", "tools-log4j.properties")
+      @log4_properties_sample_path = @sample_config_dir.join("", "tools-log4j.properties")
+      @client_properties_path      = @config_dir.join("", "client.properties")
+      @server_properties_path      = @config_dir.join("", "server.properties")
+      @server_properties_orig_path = @config_dir.join("", "server.properties.ORIG")
+
+      @keystore_dir                = @config_dir.join("", "keystore")
+      @truststore_path             = @keystore_dir.join("", "truststore.jks")
+      @keystore_path               = @keystore_dir.join("", "keystore.jks")
+
+      @messaging_yaml_sample_path  = @miq_config_dir.join("", "messaging.kafka.yml")
+      @messaging_yaml_path         = @miq_config_dir.join("", "messaging.yml")
+    end
+
+    def ask_questions
+      puts("JJV 0.0 #{File.basename(__FILE__)} / #{__method__}")
+
+      return false unless valid_environment?
+      ask_for_parameters
+      show_parameters
+      return false unless agree("\nProceed? (Y/N): ")
+      return false unless valid_parameters?(@hostname)
+      true
+    end
+
+    def activate
+      puts("JJV 0.0 #{File.basename(__FILE__)} / #{__method__}")
+      begin
+        configure_log3               ## STEP 3: Create the log4 configuration file
```

Comment:
minor but the method is named log3, but the feature is log4?

Reply:
renamed to create_tools_logconfig